tests/core/modules/test_loss.py

```diff
@@ -0,0 +1,234 @@
+    pred, target = forces
+    ref_norm = torch.linalg.vector_norm(pred - target, ord=2, dim=-1)
+    assert torch.allclose(loss(pred, target, natoms), ref_norm)
+
+
+def test_energy_mae_reduction(energy, natoms):
+    # this is testing on a single process i.e. world_size=1
+    # mean reduction
+    loss = DDPLoss(loss_name="mae", reduction="mean")
+    ref_loss = nn.L1Loss(reduction="mean")
+    pred, target = energy
+    assert torch.allclose(loss(pred, target, natoms), ref_loss(pred, target))
+    # sum reduction
+    loss = DDPLoss(loss_name="mae", reduction="sum")
+    ref_loss = nn.L1Loss(reduction="sum")
+    assert torch.allclose(loss(pred, target, natoms), ref_loss(pred, target))
+
+
+def test_stress_mae_reduction(anisotropic_stress, natoms):
+    # this is testing on a single process i.e. world_size=1
+    # mean reduction
+    loss = DDPLoss(loss_name="mae", reduction="mean")
+    ref_loss = nn.L1Loss(reduction="mean")
+    pred, target = anisotropic_stress
+    assert torch.allclose(loss(pred, target, natoms), ref_loss(pred, target))
+    # sum reduction
+    loss = DDPLoss(loss_name="mae", reduction="sum")
+    ref_loss = nn.L1Loss(reduction="sum")
+    assert torch.allclose(loss(pred, target, natoms), ref_loss(pred, target))
+
+
+def test_l2norm_reduction(forces, natoms):
+    # this is testing on a single process i.e. world_size=1
+    # mean reduction
+    loss = DDPLoss(loss_name="l2norm", reduction="mean")
+    pred, target = forces
+    ref_norm = torch.linalg.vector_norm(pred - target, ord=2, dim=-1)
+    ref_loss = ref_norm.mean()
+    assert torch.allclose(loss(pred, target, natoms), ref_loss)
+    # sum reduction
+    loss = DDPLoss(loss_name="l2norm", reduction="sum")
+    ref_loss = ref_norm.sum()
+    assert torch.allclose(loss(pred, target, natoms), ref_loss)
+
+
+def test_mse_reduction(forces, natoms):
+    # this is testing on a single process i.e. world_size=1
+    # mean reduction
+    loss = DDPLoss(loss_name="mse", reduction="mean")
+    ref_loss = nn.MSELoss(reduction="mean")
+    pred, target = forces
+    assert torch.allclose(loss(pred, target, natoms), ref_loss(pred, target))
+    # sum reduction
+    loss = DDPLoss(loss_name="mse", reduction="sum")
+    ref_loss = nn.MSELoss(reduction="sum")
+    assert torch.allclose(loss(pred, target, natoms), ref_loss(pred, target))
+
+
+def split_batch_for_ddp(
+    task: str, pred: torch.Tensor, target: torch.Tensor, natoms: torch.Tensor
+):
+    if task == "energy":
+        return list(torch.split(pred, 1)), list(torch.split(target, 1))
+    elif task == "forces":
+        split_shape = natoms.tolist()
+        return list(torch.split(pred, split_shape, dim=0)), list(
+            torch.split(target, split_shape, dim=0)
+        )
+    else:
+        raise ValueError(f"Invalid task: {task}")
+
+
+def run_ddp_loss(pred, target, natoms, loss_name, reduction):
+    loss = DDPLoss(loss_name=loss_name, reduction=reduction)
+    local_rank = os.environ["LOCAL_RANK"]
```

Comment:
nit: i know i said to do this but i forgot its better to use distutils.get_rank() instead (local_rank technically means the rank on the current node, but doesnt really matter here since its a test)